psw/ae/aesm_service/source/aesm/application/main.cpp

```diff
@@ -73,8 +73,12 @@ void signal_handler(int sig)
     }
 }
 
-int main() {
-    if(daemon(0, 0) < 0)
+int main(int argc, char *argv[]) {
+    // The only command line option that is supported is --no-daemon,
+    // so for now we can hardcode a check against argv[1], which will
+    // be NULL if no argument is provided.
+    bool noDaemon = (strcmp(argv[1], "--no-daemon") == 0);
```

Comment:
OK for strcmp.  Still must check argc before using argv.  aesm_service without any arg will coredump since arg[1] is NULL.